src/extractor/scripting_environment_lua.cpp

```diff
@@ -18,18 +19,70 @@
 #include <luabind/tag_function.hpp>
 
 #include <osmium/osm.hpp>
+#include <osmium/visitor.hpp>
 
 #include <tbb/parallel_for.h>
 
 #include <memory>
 #include <sstream>
 
+#include <stdio.h>
+
 namespace osrm
 {
 namespace extractor
 {
+
+class CacheHandler : public osmium::handler::Handler {
+  public:
+    explicit CacheHandler(){};
+
+    void node(const osmium::Node &node)
+    {
+      std::cout << "node id  : " << node.id() << std::endl;
+    };
+};
+
+// location_handler_type* LuaScriptingEnvironment::location_handler;
+
+void LuaScriptingEnvironment::setCache(const std::string &file_name)
+{
+    // Node Location Cache
+    std::cout << "Create node cache from file " << file_name <<  std::endl;
+
+    const osmium::io::File input_file(file_name);
+    osmium::io::Reader cacheReader(input_file, osmium::osm_entity_bits::node);
+    int fd = open("nodes.cache", O_RDWR | O_CREAT, 0666);
+    if (fd == -1)
+    {
+	throw std::runtime_error(strerror(errno));
+    }
+
+    index_type index{fd};
+    index_neg_type index_neg;
```

Comment:
Your indices go out of scope after the `setCache` function returns. That's the reason once yo run the [extractor here](https://github.com/Project-OSRM/osrm-backend/pull/3223/files#diff-38493d025685d69c5109f547b722a4ebR164) your locations are no longer valid.